tests/components/smartthings/test_config_flow.py

```diff
@@ -4,207 +4,266 @@
+    """Test import returns user."""
     flow = SmartThingsFlowHandler()
     flow.hass = hass
-    result = await flow.async_step_user()
+
+    result = await flow.async_step_import()
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
     assert result["step_id"] == "user"
+    assert result["description_placeholders"][
+        "webhook_url"
+    ] == smartapp.get_webhook_url(hass)
 
 
-async def test_step_init(hass):
-    """Test the access token form is shown for an init flow."""
+async def test_step_user(hass):
+    """Test the webhook confirmation is shown."""
     flow = SmartThingsFlowHandler()
     flow.hass = hass
-    result = await flow.async_step_import()
+
+    result = await flow.async_step_user()
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
     assert result["step_id"] == "user"
+    assert result["description_placeholders"][
+        "webhook_url"
+    ] == smartapp.get_webhook_url(hass)
 
 
-async def test_base_url_not_https(hass):
-    """Test the base_url parameter starts with https://."""
+async def test_step_user_aborts_invalid_webhook(hass):
+    """Test flow aborts if webhook is invalid."""
     hass.config.api.base_url = "http://0.0.0.0"
     flow = SmartThingsFlowHandler()
     flow.hass = hass
-    result = await flow.async_step_user({"access_token": str(uuid4())})
 
-    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "user"
-    assert result["errors"] == {"base": "base_url_not_https"}
+    result = await flow.async_step_user()
 
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "invalid_webhook_url"
+    assert result["description_placeholders"][
+        "webhook_url"
+    ] == smartapp.get_webhook_url(hass)
+    assert "component_url" in result["description_placeholders"]
 
-async def test_invalid_token_format(hass):
-    """Test an error is shown for invalid token formats."""
+
+async def test_step_user_advances_to_pat(hass):
+    """Test user step advances to the pat step."""
     flow = SmartThingsFlowHandler()
     flow.hass = hass
-    result = await flow.async_step_user({"access_token": "123456789"})
+
+    result = await flow.async_step_user({})
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "user"
-    assert result["errors"] == {"access_token": "token_invalid_format"}
+    assert result["step_id"] == "pat"
 
 
-async def test_token_already_setup(hass):
-    """Test an error is shown when the token is already setup."""
+async def test_step_pat(hass):
+    """Test pat step shows the input form."""
     flow = SmartThingsFlowHandler()
     flow.hass = hass
+
+    result = await flow.async_step_pat()
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "pat"
+    assert result["errors"] == {}
+    assert result["data_schema"]({CONF_ACCESS_TOKEN: ""}) == {CONF_ACCESS_TOKEN: ""}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5359319)